src/modules.js

```diff
@@ -1,17 +1,39 @@
+import axios from "axios";
+import Bowser from "bowser";
+import firebase from "firebase/app";
+import 'firebase/auth';
 /* global __analytics */
 // Note there has been special combination of if statement applied here so the unused code can be eliminated
 // during dead code elimination process, while editing/refactoring this file please take care,
 // so that the elimination process still works.
 // Also note that each analytics cannot be split into separate files, as it will break code elimination.
 const analyticModules = [];
-
+const filterProperties = (allowProperties = [], disallowProperties = [], properties) => {
+  let eventProperties = properties;
+  if(allowProperties.length > 0 ) {
+    eventProperties =  Object.keys(eventProperties)
+      .reduce((object, key) => (allowProperties.includes(key)
+          ? {...object, [key]: eventProperties[key]}
+          : object
+      ), {});
+  }
+  if(disallowProperties.length > 0 ) {
+    eventProperties =  Object.keys(eventProperties)
+      .reduce((object, key) => (disallowProperties.includes(key)
+          ? object
+          : {...object, [key]: eventProperties[key]}
+      ), {});
+  }
+  return eventProperties;
+}
 
 // --- Debug module ---
-if (__analytics.debug.isEnabled) {
+if (__analytics.debug?.isEnabled) {
   const debugModule = function () {
     console.log('Analytics (initialization)');
 
     return {
+      name: 'debug',
```

Comment:
H: I would suggest to create Enum with analytic names and here use the value from the enum object, as well as in the application code use the enum values rather then repeating the strings by hand.